exercises/resistor-color-trio/canonical-data.json

```diff
@@ -0,0 +1,46 @@
+{
+  "exercise": "resistor-color-trio",
+  "version": "1.0.0",
+  "cases": [
+    {
+      "description": "Orange and orange and black",
+      "property": "description",
+      "input": {
+        "colors": ["orange", "orange", "black"]
+      },
+      "expected": "The value of this resistor is 33 Ohms."
```

Comment:
Would it sufficient to just expect `33 Ohms`?  The rest of the sentence seems unnecessary.

Reply:
I like when programs talk in real sentences. So, just returning the value is too little imo. But I agree that the sentence is adding noise... 

I see 2 options: either a shorter sentence, like "This is a 33 Ohms resistor". Or "The value of orange-orange-black is 33 Ohms". 
That latter would introduce a second iteration on `colors`. I'm not sure if that adds something interesting.
@ErikSchierboom I don't mind going with the shortened sentence. Do you?

Reply:
For the problem specification, I'd prefer if the expected value were the integer `33` instead of any kind of string.

String formatting is not integral to this exercise, and a trivial wrapper function makes a sentence out of an integer. The reverse operation is less simple. 

We should be teaching good practices. One good practice is to avoid conflating two different logical operations into one function. Computing a resistor's value from its colors and formatting a human-friendly string describing a resistor are two different logical operations.

Reply:
> For the problem specification, I'd prefer if the expected value were the integer `33` instead of any kind of string.

I’d strongly second that point. The first two exercises are focused on calculating the integer value by parsing a string, and this adds on the complexity of that from the previous two. Adding string interpolation is unnecessary and it’s not like we’re lacking in exercises for that anyway. What we have been lacking is _simple_ exercises that gradually expand on integer maths.

The very debate about preference about what string to return displays another problem with having one function that addresses two concerns; the string you’d print would be culturally and locale sensitive in the real world, so you’d want one function doing the math and another doing the interpolation. If we don’t want to add a second function, then _should_ we make _another_ exercise that spends time on English-specific string formatting esoterica (ie 3000 > 3K)?

Reply:
I like the concept of separating the two concerns, one calculating the values, and one for returning a string.
I do think we need the string part of this exercise. Not the interpolation, but for the part that is about Ohms vs kilo Ohms. 
That is also meant as a bridge to the next Resistor exercise(s), where the Ohms value is combined with other values in other units (like a percentage for accuracy).
My proposal would be: separate the concerns in 2 methods, one for calculation, one for the ‘report’/printing. Can we all live with that?

Reply:
> should we make another exercise that spends time on English-specific string formatting esoterica (ie 3000 > 3K)
Agreed. And IIRC there’s an issue to create such an exercise.

Reply:
Sure, a two-method solution works for me. One accepts as input a list of colors and returns an integer; the other accepts an integer and returns a string. We should test them independently.

Reply:
> I like the concept of separating the two concerns, one calculating the values, and one for returning a string.

Fine with me!  

> One accepts as input a list of colors and returns an integer; the other accepts an integer and returns a string.

What do you think @F3PiX?

Reply:
I'm totally okay with it 👍